user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -96,14 +102,23 @@ public class SampleDetailsComponent extends PageArea implements Serializable {
   private final transient SampleDetailsComponentHandler sampleDetailsComponentHandler;
   private final List<ValueChangeListener<ComponentValueChangeEvent<TextField, String>>> searchFieldListeners = new ArrayList<>();
   private Context context;
+  private final ExperimentInformationService experimentInformationService;
+  private final BatchRegistrationService batchRegistrationService;
 
   public SampleDetailsComponent(@Autowired SampleInformationService sampleInformationService,
       @Autowired BatchRegistrationService batchRegistrationService,
-      @Autowired SampleRegistrationService sampleRegistrationService) {
+      @Autowired SampleRegistrationService sampleRegistrationService,
+      @Autowired ExperimentInformationService experimentInformationService) {
+    this.experimentInformationService = experimentInformationService;
     initSampleView();
+    this.batchRegistrationService = batchRegistrationService;
+    this.sampleRegistrationService = sampleRegistrationService;
     this.sampleDetailsComponentHandler = new SampleDetailsComponentHandler(
-        sampleInformationService, batchRegistrationService,
-        sampleRegistrationService);
+        sampleInformationService, this.batchRegistrationService,
+        this.sampleRegistrationService);
+
+    Button registerButton2 = new Button("Register", this::onRegisterButtonClicked);
```

Comment:
There should only be one registerButton 
```suggestion
    Button registerButton = new Button("Register", this::onRegisterButtonClicked);
```